### ISSUE TITLE
[Bug]: Ios App Crashed only v16.6.1

### ISSUE THREAD

**Issue by mrcse** [fluttercommunity/plus_plugins#2223]
### Platform

Iphone 13

### Plugin

connectivity_plus

### Version

4.0.2

### Flutter SDK

3.13.6

### Steps to reproduce

<img width="1059" alt="Screenshot 2023-10-05 at 10 57 44 PM" src="https://github.com/fluttercommunity/plus_plugins/assets/73348512/af95856d-1ccc-4078-a127-b1fd7b8135d3">

The app crashed while launching because i call the checkConnectivity() at startup 

### Code Sample

```dart
final result = await Connectivity().checkConnectivity();
      if (result == ConnectivityResult.mobile ||
          result == ConnectivityResult.wifi) {
prrint('Success');

}
```

### Logs

```shell
https://github.com/fluttercommunity/plus_plugins/assets/73348512/af95856d-1ccc-4078-a127-b1fd7b8135d3
```

### Flutter Doctor

```shell
flutter doctor -v
[✓] Flutter (Channel stable, 3.13.6, on macOS 14.0 23A344 darwin-x64, locale en-PK)
    • Flutter version 3.13.6 on channel stable at /usr/local/Caskroom/flutter/3.7.0/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ead455963c (9 days ago), 2023-09-26 18:28:17 -0700
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/mrcse/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.82.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (3 available)
    • Jamshid’s iPhone (mobile) • 00008120-001170143AC0C01E • ios            • iOS 17.0.2 21A351
    • macOS (desktop)           • macos                     • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)              • chrome                    • web-javascript • Google Chrome 113.0.5672.126

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

**Comment by monfadev** [comment]
i have same issue

**Comment by viphoamt** [comment]
i have same issue in ios 17

**Comment by vbuberen** [comment]
Duplicate of #2216 and #2203 and a few other issues.

This is already resolved and will be released this week.
See #2169 for fix PR details or #2216 for instruction on how you can already use the fixed version.

P.S. Would be great if people could use search by issues before opening bug reports and spamming with `I have the same issue`.

**Comment by mrcse** [comment]
Thanks
